src/XmlAgent.cc

```diff
@@ -885,9 +879,9 @@ YCPValue XmlAgent::Execute(const YCPPath &path, const YCPValue& value, const YCP
     {
         filename =  fileName;
     }
-    else if (input && *input)
+    else if (input == "string")
```

Comment:
three timethis comparison. What about having it as constant and compare to that constant?